bin/cron/build_contact_rollups

```diff
@@ -0,0 +1,428 @@
+    WHERE contact_rollups.dashboard_user_id = src.teacher_id"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def mysql_multi_connection
+  # return a connection with the MULTI_STATEMENTS flag set that allows multiple statements in one DB call
+  Sequel.connect(CDO.pegasus_db_writer.sub('mysql:', 'mysql2:'), flags: ::Mysql2::Client::MULTI_STATEMENTS)
+end
+
+def update_geo_data_from_forms(form_kind)
+  start = Time.now
+  log "Updating geo data for form kind #{form_kind}"
+
+  record_count = 0
+  num_updated = 0
+  num_parse_errors = 0
+  update_batch = ""
+  conn = mysql_multi_connection
+  time_last_output = start
+
+  PEGASUS_DB[:forms].where(kind: form_kind).each do |form|
+    record_count += 1
+    begin
+      data = JSON.parse(form[:data])
+    rescue JSON::ParserError
+      # we have just a couple of records with invalid JSON (duplicate fields), rescue and skip these
+      num_parse_errors += 1
+      next
+    end
+
+    email = data['email_s'].presence
+    next if email.nil?
+
+    # get user-supplied address/location data from form if present, from any of the differently-named location fields across all form kinds
+    street_address = data['location_street_address_s'].presence || data['user_street_address_s'].presence
+    city = data['location_city_s'].presence || data['user_city_s'].presence || data['teacher_city_s'].presence
+    state = data['location_state_s'].presence || data['teacher_state_s'].presence
+    if state.nil?
+      # note that the 'user_state_s' record is in fact a *state code*, not a state name
+      state_code = data['location_state_code_s'].presence || data['state_code_s'].presence || data['user_state_s'].presence
+      state = get_us_state_from_abbr(abbr: state_code, include_dc: true).presence unless state_code.nil?
+    end
+    postal_code = data['location_postal_code_s'].presence || data['zip_code_s'].presence || data['user_postal_code_s'].presence
+    country = data['location_country_s'].presence || data['country_s'].presence || data['teacher_country_s'].presence
+
+    # skip if this form data has no address/location info at all
+    next if street_address.nil? && city.nil? && state.nil? && postal_code.nil? && country.nil?
+
+    # Truncate all fields to 255 chars. We have some data longer than 255 chars but it is all garbage that somebody typed.
+    street_address = street_address[0...255] unless street_address.nil?
+    city = city[0...255] unless city.nil?
+    state = state[0...255] unless state.nil?
+    postal_code = postal_code[0...255] unless postal_code.nil?
+    country = country[0...255] unless country.nil?
+
+    # add to batch to update
+    update_batch += conn[:contact_rollups].where(email: email).update_sql(
+      street_address: street_address, city: city, state: state, postal_code: postal_code, country: country) + ";"
+
+    num_updated += 1
+
+    if record_count % UPDATE_BATCH_SIZE == 0
```

Comment:
Does it make more sense for this to depend on `num_updated` rather than `record_count`?